doc/devel/MEP/MEP28.rst

```diff
@@ -0,0 +1,181 @@
+-------------------------------------------------------
+
+This MEP proposes that two parameters (e.g., ``transform_in`` and
+``transform_out`` be added to the cookbook function that computes the
+statistics for the boxplot function. These will be optional keyword-only
+arguments and can easily be set to ``lambda x: x`` as a no-op when omitted
+by the user. The ``transform_in`` function will be applied to the data
+as the ``boxplot_stats`` function loops through each subset of the data
+pass to it. After the list of statistics dictionaries are computed the
+``transform_out`` function is applied to each value in the dictionaries.
+
+These transformation can then be added to the call signature of
+``Axes.boxplot`` with little impact to that method's complexity. This is
+because they can be directly passed to ``cbook.boxplot_stats``.
+Alternatively, ``Axes.boxplot`` could be modified to accept an optional
+statistical function kwarg and a dictionary of parameters to be direcly
+passed to it.
+
+At this point in the implementation users and external libraries like
+seaborn would have complete control via the ``Axes.boxplot`` method. More
+importantly, at the very least, seaborn would require no changes to its
+API to allow users to take advantage of these new options.
+
+Simiplifications to the ``Axes.boxplot`` API and other functions
+----------------------------------------------------------------
+
+Simplifying the boxplot method consists primarily of deprecating and then
+removing the redundant parameters. Optionally, a next step would include
+rectifying minor terminological inconsistencies between ``Axes.boxplot``
+and ``Axes.boxplot``.
+
+The parameters to be deprecated and removed include:
+
+  1. ``usermedians`` - processed by 10 SLOC, 3 ``if`` blocks, a ``for`` loop
+  2. ``conf_intervals`` - handled by 15 SLOC, 6 ``if`` blocks, a ``for`` loop
+  3. ``sym`` - processed by 12 SLOC, 4 ``if`` blocks
+
+Removing the ``sym`` option allows all code in handling the remaining
+styling parameters to be moved to ``Axes.bxp``. This doesn't remove
+any complexity, but does reinforce the single responsibility principle
+among ``Axes.bxp``, ``cbook.boxplot_stats``, and ``Axes.boxplot``.
+
+Additionally, the ``notch`` parameter could be renamed ``shownotches``
+to be consistent with ``Axes.bxp``. This kind of cleanup could be taken
+a step further and the ``whis``, ``bootstrap``, ``autorange`` could
+be rolled into the kwargs passed to the new ``statfxn`` parameter.
+
+Backward compatibility
+======================
+
+Implementation of this MEP would eventually result in the backwards
+incompatible deprecation and then removeal of the keyword parameters
+``usermedians``, ``conf_intervals``, and ``sym``. Curosry searches on
+GitHub indicated that ``usermedians``, ``conf_intervals`` are used by
+few users, who all seem to have a very strong knowledge of matplotlib.
+A robust deprecation cycle should provide sufficient time for these
```

Comment:
Considering we are have discussion about one release cycle versus two release cycle on the finance module right now, it might be worth being specific here.